src/mds/MDSMap.h

```diff
@@ -314,7 +314,12 @@ class MDSMap {
   int64_t get_first_data_pool() const { return *data_pools.begin(); }
   int64_t get_metadata_pool() const { return metadata_pool; }
   bool is_data_pool(int64_t poolid) const {
-    return std::binary_search(data_pools.begin(), data_pools.end(), poolid);
+    std::vector<int64_t>::const_iterator b = data_pools.begin();
+    std::vector<int64_t>::const_iterator e = data_pools.end();
+    std::vector<int64_t>::const_iterator p = std::find(b, e, poolid);
```

Comment:
Good catch!

Above three lines can be simplified into
"auto p = std::find(data_pools.begin(), data_pools.end(), poolid);"